src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -727,30 +728,14 @@ Namespace Microsoft.CodeAnalysis.Operations
         End Function
 
         Private Function CreateBoundCollectionInitializerExpressionOperation(boundCollectionInitializerExpression As BoundCollectionInitializerExpression) As IObjectOrCollectionInitializerOperation
-            Dim initializers As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(Function() boundCollectionInitializerExpression.Initializers.SelectAsArray(Function(n) CreateBoundCollectionElementInitializerOperation(n)))
+            Dim initializers As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(Function() boundCollectionInitializerExpression.Initializers.SelectAsArray(Function(n) Create(n)))
             Dim syntax As SyntaxNode = boundCollectionInitializerExpression.Syntax
             Dim type As ITypeSymbol = boundCollectionInitializerExpression.Type
             Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundCollectionInitializerExpression.ConstantValueOpt)
             Dim isImplicit As Boolean = boundCollectionInitializerExpression.WasCompilerGenerated
             Return New LazyObjectOrCollectionInitializerExpression(initializers, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
-        Private Function CreateBoundCollectionElementInitializerOperation(boundExpression As BoundExpression) As IOperation
```

Comment:
>CreateBoundCollectionElementInitializerOperation [](start = 25, length = 48)

Do we have any tests covering late bound invocation of the Add method? Didn't notice any among changed tests, but I could miss something.

Reply:
Adding one, but it is blocked by [https://github.com/dotnet/roslyn/issues/27034](https://github.com/dotnet/roslyn/issues/27034). Will fix in a separate PR.

---
In reply to: [189345284](https://github.com/dotnet/roslyn/pull/26781#discussion_r189345284) [](ancestors = 189345284)